src/app/work-item/work-item-iteration-association-modal/work-item-iteration-association-modal.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, ViewChild, OnInit, Output, EventEmitter, Input, OnChanges } from '@angular/core';
+
+import { AuthenticationService } from '../../auth/authentication.service';
+import { Broadcaster } from '../../shared/broadcaster.service';
+import { Logger } from '../../shared/logger.service';
+
+import { WorkItem }          from '../../models/work-item';
+import { WorkItemService }   from '../work-item.service';
+import { IterationModel }    from '../../models/iteration.model';
+import { IterationService }  from '../../iteration/iteration.service';
+
+
+@Component({
+  selector: 'fab-planner-associate-iteration-modal',
+  templateUrl: './work-item-iteration-association-modal.component.html',
+  styleUrls: ['./work-item-iteration-association-modal.component.scss']
+})
+export class FabPlannerAssociateIterationModalComponent implements OnInit, OnChanges {
+
+  @Input() workItem: WorkItem;
+  @ViewChild('dropdownButton') dropdownButton: any;
+  @ViewChild('iterationAssociationModal') iterationAssociationModal: any;
+
+  iterations: IterationModel[];
+  selectedIteration: IterationModel;
+
+  constructor(
+    private auth: AuthenticationService,
+    private broadcaster: Broadcaster,
+    private workItemService: WorkItemService,
+    private logger: Logger,
+    private iterationService: IterationService,
+  ) {}
+
+  ngOnInit() {
+    this.iterations = this.iterationService.iterations;
+  }
+
+  ngOnChanges() {
+
+  }
+
+  onChangeIteration(iteration: IterationModel): void {
+    // Cache changed iteration
+    this.selectedIteration = iteration;
+
+    // Update dropdown button content
+    this.dropdownButton.nativeElement.innerHTML = this.selectedIteration.attributes.name + ' <span class="caret"></span>';
+
+    // Enable/disable confirmation button if iteration was changed
+    let commit = document.querySelector("#associate-iteration-button");
+    if (this.workItem.relationalData.iteration.id === this.selectedIteration.id) {
```

Comment:
In case there is no iteration associated with the item, this.workItem.relationalData.iteration.id will throw an error.